//    Copyright 2022, farm-ng inc.
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.

syntax = "proto3";

import "farm_ng/core/linalg.proto";
import "farm_ng/core/lie.proto";

package farm_ng.core.proto;

// A position and orientation for an object or reference frame in 3D space
message Pose {
  // point_in_b = b_from_a * point_in_a
  Isometry3F64 a_from_b = 1;
  string frame_a = 2;
  string frame_b = 3;


  // linear velocity of body a (with respect to body b) in frame b.
  Vec3F64 linvel_of_a_in_b = 4;

  // angular velocity of body a (with respect to body b) in frame a.
  Vec3F64 angvel_of_a_in_a = 5;
}

// A sequence of poses
message PoseTree {
  repeated Pose poses = 1;
}
